tailwind.config.js

```diff
@@ -173,7 +482,63 @@ module.exports = {
         },
       };
 
-      addUtilities(utilities, ["responsive"]);
+      addUtilities(utilities, ["lib", "DEFAULT", "responsive"]);
+    }),
+
+    /* =========================================================================
+      Variants
+      ========================================================================== */
+    plugin(function ({ addVariant, e }) {
+      addVariant("lib", ({ container, separator, modifySelectors }) => {
+        modifySelectors(({ selector }) => {
+          return buildSelectorVariant(selector, "lib", separator, message => {
+            throw container.error(message);
+          });
+        });
+      });
+      addVariant("lib:hover", generatePseudoClassVariant("hover"));
+      addVariant(
+        "lib:focus-within",
+        generatePseudoClassVariant("focus-within"),
+      );
+      addVariant("lib:focus", generatePseudoClassVariant("focus"));
+      addVariant("lib:disabled", generatePseudoClassVariant("disabled"));
+
+      function generateDataClassVariant(dataName, dataBool, variant) {
+        // lib:aria-selected OR aria-selected
+        const variantName = variant ? `${variant}:${dataName}` : dataName;
+
+        addVariant(variantName, ({ modifySelectors, separator }) => {
+          modifySelectors(({ className }) => {
+            const selector = selectorClass => {
+              // `.${e(`is-range-selection${separator}${className}`)}[data-is-range-selection]`
+              if (dataBool)
+                return `.${e(`${selectorClass}`)}[data-${dataName}]`;
+              // `.${e(`aria-selected${separator}${className}`)}[aria-selected="true"]`
+              return `.${e(`${selectorClass}`)}[${dataName}="true"]`;
+            };
+            // `aria-selected${separator}${className}`
+            const dataClass = `${dataName}${separator}${className}`;
+            // `lib${separator}aria-selected${separator}${className}`
+            const variantDataClass = variant
+              ? `${variant}${separator}${dataClass}`
+              : dataClass;
+
+            return selector(variantDataClass);
+          });
+        });
+      }
```

Comment:
Can we clean this up a just a bit?

```js
        addVariant(variantName, ({ modifySelectors, separator }) => {
          modifySelectors(({ className }) => {
            const selector = selectorClass => {
              if (dataBool) {
                return `.${e(`${selectorClass}`)}[data-${dataName}]`;
              }
              return `.${e(`${selectorClass}`)}[${dataName}="true"]`;
            };

            const dataClass = `${dataName}${separator}${className}`;
            const variantDataClass = variant
              ? `${variant}${separator}${dataClass}`
              : dataClass;

            return selector(variantDataClass);
          });
        });
      }
```

Reply:
Oh, I thought the comments will help us whats happening since everything is now within `string literal`

Reply:
> Oh, I thought the comments will help us whats happening since everything is now within string literal

yes but it's kinda messy. i would rather like to see actual string values instead of the same template interpolations for example

// [aria-selected=true]